### PR TITLE
feat: Extract proof_data_handler into separate crate

### PR DESCRIPTION
**Pull request by AnastasiiaVashchuk**
## What ❔

Extract `zksync_core::proog_data_handler` to separate  crate in the `node` folder.

## Why ❔

Part of the "modularize the codebase" process, required for legolizer and publishing on crates.io.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/zksync_core/src/lib.rs [hunk 1]
```diff
@@ -782,7 +782,7 @@ pub async fn initialize_components(
     }
 
     if components.contains(&Component::ProofDataHandler) {
-        task_futures.push(tokio::spawn(proof_data_handler::run_server(
+        task_futures.push(tokio::spawn(run_server(
```
Comment (slowli) [inline core/lib/zksync_core/src/lib.rs:785]: Bikeshedding: IMO, using a qualified name here (i.e., `zksync_proof_data_handler::run_server(..)` would make more sense here; `run_server` is quite ambiguous (lots of components that have their servers, and `lib.rs` concerns all components).

File: core/node/proof_data_handler/README.md
Comment (slowli) [inline core/node/proof_data_handler/README.md:1]: Is this file supposed to be empty? I think it should be possible to insert a couple of sentences with the component description 🙃

File: core/node/proof_data_handler/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+keywords.workspace = true
+categories.workspace = true
+
+[dependencies]
+zksync_config.workspace = true
+zksync_dal.workspace = true
+zksync_object_store.workspace = true
+zksync_prover_interface.workspace = true
+zksync_types.workspace = true
+zksync_utils.workspace = true
```
Comment (slowli) [inline core/node/proof_data_handler/Cargo.toml:18]: Nit: AFAICT, this dependency can be avoided. `zksync_utils` is only used for a single `u256_to_h256` conversion, which could be replaced with `H256::from_low_u64_be(2)` (please test that the produced `H256` are indeed equal).

File: core/node/proof_data_handler/README.md
Comment (AnastasiiaVashchuk) [reply core/node/proof_data_handler/README.md:1]: nope
i have opened PR but  there still has to be some commits
that why I did not request review
will add now

**Review by slowli — APPROVED** [review]
Nice.

**Comment by popzxc** [comment]
@AnastasiiaVashchuk this PR seems to have merge conflicts now, could you resolve them please?

**Review by popzxc — APPROVED** [review]

**Pull request merged**